core/aws-core/src/main/java/software/amazon/awssdk/awscore/internal/authcontext/AuthorizationStrategyFactory.java

```diff
@@ -64,7 +65,8 @@ private TokenAuthorizationStrategy tokenAuthorizationStrategy() {
 
     private AwsCredentialsAuthorizationStrategy awsCredentialsAuthorizationStrategy() {
         Signer defaultSigner = clientConfiguration.option(SdkAdvancedClientOption.SIGNER);
-        AwsCredentialsProvider defaultCredentialsProvider = clientConfiguration.option(AwsClientOption.CREDENTIALS_PROVIDER);
+        IdentityProvider<? extends AwsCredentialsIdentity> defaultCredentialsProvider =
+            clientConfiguration.option(AwsClientOption.CREDENTIALS_IDENTITY_PROVIDER);
```

Comment:
If CREDENTIALS_IDENTITY_PROVIDER is null, we should fall back to CREDENTIALS_PROVIDER, in case they're using an old client version with newer core.

Reply:
Discussed offline - fallback not needed. Don't think child classes change the provider.

Reply:
Confirmed after generated all AWS services, that nothing references AwsClientOption.CREDENTIALS_PROVIDER. Only usage of that is in AwsDefaultClientBuilder.finalizeChildConfiguration to set this option for older generated clients. So confirmed that no fallback is needed in the AuthorizationStratergyFactory, so removing it.